src/Control/Middleware/HTTPCacheControlMiddleware.php

```diff
@@ -0,0 +1,561 @@
+<?php
+
+namespace SilverStripe\Control\Middleware;
+
+use InvalidArgumentException;
+use SilverStripe\Control\HTTP;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Control\HTTPResponse_Exception;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Core\Resettable;
+
+class HTTPCacheControlMiddleware implements HTTPMiddleware, Resettable
```

Comment:
I guess there are a couple points here

1. It's namespaced, so even if the classname is the same it will actually be different (though I accept upgrading much easier as you just need the class import once and that's it).
2. Our convention appears to dictate that we add the "thing" that the class is to the class name (and not only the namespace) so you know what it is just by looking at it.
3. If people really want to upgrade their code by *only* adding a class import they can just do `use ... as HTTPCacheControl`